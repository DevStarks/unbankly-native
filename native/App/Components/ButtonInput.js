import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, TouchableOpacity } from 'react-native'
import Text from './Text'
import styles from './Styles/ButtonInputStyles'
import ExamplesRegistry from '../Services/ExamplesRegistry'

ExamplesRegistry.addComponentExample('ButtonInput', () =>
  <ButtonInput>This is text</ButtonInput>
)

class ButtonInput extends Component {
  // For a complete list of props
  // https://facebook.github.io/react-native/docs/textinput.html#props

  static propTypes = {
    value: PropTypes.string,
    onSelect: PropTypes.func.isRequired,
    options: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.any,
        name: PropTypes.string
      })
    ).isRequired
  }

  onButtonPress (id) {
    return () => {
      this.setState({ selected: id })
      this.props.onSelect(id)
    }
  }

  renderButton (option, i) {
    const selected = this.props.value === option.id
    return (
      <TouchableOpacity
        key={i}
        activeOpacity={0.7}
        style={[styles.button, (selected ? styles.selected : {})]}
        onPress={this.onButtonPress(option.id)}>
        <Text style={styles.text}>{option.name}</Text>
      </TouchableOpacity>
    )
  }

  renderButtons () {
    return this.props.options.map(this.renderButton.bind(this))
  }

  render () {
    return (
      <View style={[styles.ButtonInput, this.props.style]}>
        {this.renderButtons()}
      </View>
    )
  }
}

export default ButtonInput
