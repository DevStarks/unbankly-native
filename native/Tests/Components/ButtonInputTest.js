// import 'react-native'
import React from 'react'
import { TouchableOpacity } from 'react-native'
import ButtonInput from '../../App/Components/ButtonInput'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

test('ButtonInput component renders correctly', () => {
  const tree = renderer.create(
    <ButtonInput
      onSelect={() => {}}
      options={[{id: 1, name: 'Option 1'}, {id: 2, name: 'Option 2'}]}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('it renders TouchableOpacity components', () => {
  const wrapper = shallow(
    <ButtonInput
      onSelect={() => {}}
      options={[{id: 1, name: 'Option 1'}]}
    />)

  expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
})

test('onSelect', () => {
  let i = null
  const onSelect = id => { i = id }
  const wrapperPress = shallow(
    <ButtonInput
      onSelect={onSelect}
      options={[{id: 1, name: 'Option 1'}, {id: 2, name: 'Option 2'}]}
    />)

  expect(i).toBe(null)
  wrapperPress.find(TouchableOpacity).first().simulate('press')
  expect(i).toBe(1)
})
