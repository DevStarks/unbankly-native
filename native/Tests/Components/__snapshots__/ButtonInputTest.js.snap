// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonInput component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F2EFEB",
        "borderColor": "#F2EFEB",
        "borderWidth": 1,
        "flex": 1,
        "marginHorizontal": 1,
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#637381",
            "fontFamily": "Oswald-Light",
            "fontSize": 20,
          },
          Object {
            "color": "#637381",
            "fontFamily": "Oswald-Bold",
            "fontSize": 20,
            "textAlign": "center",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      Option 1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F2EFEB",
        "borderColor": "#F2EFEB",
        "borderWidth": 1,
        "flex": 1,
        "marginHorizontal": 1,
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#637381",
            "fontFamily": "Oswald-Light",
            "fontSize": 20,
          },
          Object {
            "color": "#637381",
            "fontFamily": "Oswald-Bold",
            "fontSize": 20,
            "textAlign": "center",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      Option 2
    </Text>
  </View>
</View>
`;
